miprometheus/problems/video_text_to_class/cog/cog.py

```diff
@@ -0,0 +1,349 @@
+ #!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""video_text_to_class_problem.py: abstract base class for sequential VQA problems."""
+
+import torch
+import gzip
+import json
+import os
+import numpy as np
+
+from miprometheus.utils.data_dict import DataDict
+from miprometheus.problems.video_text_to_class.video_text_to_class_problem import VideoTextToClassProblem
+from miprometheus.problems.video_text_to_class.cog.cog_utils import train_utils as tu
+
+class COGDataset(VideoTextToClassProblem):
+	"""
+	The COG dataset is a sequential VQA dataset. Inputs are a sequence of images of simple shapes and characters on a black \
+ 	background, and a question based on these objects that relies on memory which has to be answered at every step of the \
+	sequence.
+
+	"""
+
+	def __init__(self, params):
+		"""
```

Comment:
The description is perfect, please also comment params

:param param: bla bla bla